src/plugins/discover/public/application/helpers/resolve_index_pattern.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { i18n } from '@kbn/i18n';
+import { IUiSettingsClient, SavedObject, ToastsStart } from 'kibana/public';
+import { IndexPattern } from '../../kibana_services';
+import { IndexPatternsService, SearchSource } from '../../../../data/common';
+
+type IndexPatternSavedObject = SavedObject & { title: string };
+
+interface IndexPatternData {
+  /**
+   * List of existing index patterns
+   */
+  list: IndexPatternSavedObject[];
+  /**
+   * Loaded index pattern (might be default index pattern if requested was not found)
+   */
+  loaded: IndexPattern;
+  /**
+   * Id of the requested index pattern
+   */
+  stateVal: string;
+  /**
+   * Determines if requested index pattern was found
+   */
+  stateValFound: boolean;
+}
+
+export function findIndexPatternById(
+  indexPatterns: IndexPatternSavedObject[],
+  id: string
+): IndexPatternSavedObject | undefined {
+  if (!Array.isArray(indexPatterns) || !id) {
+    return;
+  }
+  return indexPatterns.find((o) => o.id === id);
+}
+
+/**
+ * Checks if the given defaultIndex exists and returns
+ * the first available index pattern id if not
+ */
+export function getFallbackIndexPatternId(
+  indexPatterns: IndexPatternSavedObject[],
+  defaultIndex: string = ''
+): string {
+  if (defaultIndex && findIndexPatternById(indexPatterns, defaultIndex)) {
+    return defaultIndex;
+  }
+  return !indexPatterns || !indexPatterns.length || !indexPatterns[0].id ? '' : indexPatterns[0].id;
```

Comment:
100% yes, the simpler the code, the better. If you understand it after getting up without a cup of coffee, it is fine code.
Do you think this would work for?

```
return indexPatterns[0]?.id ? indexPatterns[0].id : '';
```

Reply:
That's my understanding how this should work. Think you'll need another check for `indexPatterns` though, so something like:
`return (indexPatterns && indexPatterns[0]?.id) ? indexPatterns[0].id : '';`

Reply:
thx & done